### PR TITLE
Add AVC EncodedVideoChunk definitions.

### PR DESCRIPTION
**Pull request by sandersdan**

### CODE REVIEW

**Review by chcunningham — COMMENTED** [review]
Nice! Just minor edits.

File: avc_codec_registration.src.html [hunk 1]
```diff
@@ -15,8 +15,10 @@
 
 Abstract: This registration is entered into the [[webcodecs-codec-registry]].
     It describes, for AVC (H.264), the (1) fully qualified codec strings, (2)
-    the {{VideoDecoderConfig.description}} bytes, and (3) the
-    codec-specific extensions to the {{VideoEncoderConfig}}.
+    the {{VideoDecoderConfig.description}} bytes, (3) the codec-specific
+    extensions to {{VideoEncoderConfig}}, (4) the
```
[the remark this answers is not in the document]
Comment (sandersdan) [reply avc_codec_registration.src.html:19]: The question here isn't about link syntax, it's about creating cross-references. In any case, I found that I can use for= syntax in pre.anchors to be able to create for-scoped references and used that to unify the link syntax.

File: avc_codec_registration.src.html [hunk 2]
```diff
@@ -164,6 +169,38 @@
...
+NOTE: An access unit contains exactly one primary coded picture.
+
+If the bitstream is in "avc" format, {{EncodedVideoChunk.[[internal data]]}}
+must be in canonical format, as defined in [iso14496-15] section 5.3.2.
+
+If the bitstream is in "annexb" format, {{EncodedVideoChunk.[[internal data]]}}
+must be in Annex B format, as defined in [ITU-T-REC-H.264] Annex B.
+
+NOTE: Since {{EncodedVideoChunk.[[internal data]]}} is inherently byte-aligned,
+    implementations are not required to recover byte-alignment.
```
[the remark this answers is not in the document]
Comment (sandersdan) [reply avc_codec_registration.src.html:191]: Annex B format is a bitstream format, it assumes that you start reading a stream at an arbitrary bit position, and has a process for finding the alignment of NAL headers. This note just points out that implementations will expect NAL headers to be aligned to byte boundaries, which is always expected when Annex B is transported in as a bytestream.

[the remark this answers is not in the document]
Comment (aboba) [reply avc_codec_registration.src.html:191]: Question: are the "must" statements normative? I assume they are.

**Review by aboba — COMMENTED** [review]
"If the bitstream is in {{AvcBitstreamFormat/avc}} format, no parameter sets are required.

NOTE: If the bitstream is in {{AvcBitstreamFormat/avc}} format, parameter sets necessary for decoding must be included in    {{VideoDecoderConfig.description}}."

Is this a typo?

**Comment by chcunningham** [comment]
> "If the bitstream is in {{AvcBitstreamFormat/avc}} format, no parameter sets are required.
> 
> NOTE: If the bitstream is in {{AvcBitstreamFormat/avc}} format, parameter sets necessary for decoding must be included in {{VideoDecoderConfig.description}}."
> 
> Is this a typo? "not required" and "must" are both (contradictory) normative statements.

Editors call: read to fast. The intent is that one describes bitstream format annexb, the other bitstream format avc.

**Comment by sandersdan** [comment]
> Editors call: read to fast. The intent is that one describes bitstream format annexb, the other bitstream format avc.

It's a bit more subtle. In the first sentence I say that "required" parameter sets must be included in the chunk, and then I follow up by saying which parameter sets are "required" for avc and annexb. No sets are required for avc because avc format doesn't put parameter sets in the chunks. It's still required to put the parameter sets in the description.

The requirements could be split into two separate but similar paragraphs so that "required" parameter sets doesn't need to be defined, but I didn't find a way to phrase that that seemed more clear to me.

File: avc_codec_registration.src.html [hunk 3]
```diff
@@ -164,6 +169,38 @@
...
+
+{{EncodedVideoChunk.[[internal data]]}} must be an access unit as defined in
+[[ITU-T-REC-H.264]] section 7.4.1.2.
+
+NOTE: An access unit contains exactly one primary coded picture.
+
+If the bitstream is in "avc" format, {{EncodedVideoChunk.[[internal data]]}}
+must be in canonical format, as defined in [iso14496-15] section 5.3.2.
+
+If the bitstream is in "annexb" format, {{EncodedVideoChunk.[[internal data]]}}
```
[the remark this answers is not in the document]
Comment (chcunningham) [reply avc_codec_registration.src.html]: Not done?

File: avc_codec_registration.src.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (chcunningham) [reply avc_codec_registration.src.html:191]: They are, but they probably shouldn't be. Per https://www.w3.org/Guide/manual-of-style/#N-I-sections, it seems we should find different words given the desire that the registry remain non-normative. I've started a thread with internal W3c experts. Stand by.

**Review by chcunningham — COMMENTED** [review]
Generally looks good.

**Comment by chcunningham** [comment]
> It's a bit more subtle. In the first sentence I say that "required" parameter sets must be included in the chunk, and then I follow up by saying which parameter sets are "required" for avc and annexb. No sets are required for avc because avc format doesn't put parameter sets in the chunks. It's still required to put the parameter sets in the description.
> 
> The requirements could be split into two separate but similar paragraphs so that "required" parameter sets doesn't need to be defined, but I didn't find a way to phrase that that seemed more clear to me.

IMO the wording is reasonably clear.

**Comment by chcunningham** [comment]
Editors call: Generally looks good, waiting on guidance re: normative language.

File: avc_codec_registration.src.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (sandersdan) [reply avc_codec_registration.src.html]: I think you were looking at an older version.

File: avc_codec_registration.src.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (sandersdan) [reply avc_codec_registration.src.html:189]: I've updated all of the phrasing, PTAL.

File: avc_codec_registration.src.html [hunk 4]
```diff
@@ -154,16 +160,53 @@
...
+================================================
+
+{{EncodedVideoChunk}} [=EncodedVideoChunk/[[internal data]]=] is expected to be
+an access unit as defined in [[ITU-T-REC-H.264]] section 7.4.1.2.
+
+NOTE: An access unit contains exactly one primary coded picture.
+
+If the bitstream is in {{AvcBitstreamFormat/avc}} format,
+[=EncodedVideoChunk/[[internal data]]=] is assumed to be in canonical format, as
+defined in [[iso14496-15]] section 5.3.2.
```
Comment (sandersdan) [inline avc_codec_registration.src.html:182]: I'm not totally happy with this part, "in avc/annexb format" is basically a synonym for the format of `[[internal data]]`, but this is how the two are being linked together. I'm open to suggestions that would clear up this duplication.

Comment (chcunningham) [reply avc_codec_registration.src.html:182]: I've failed to come up with something more concise. You could say "If the bitstream is configured to be avc format, ..." - maybe slightly improves it? I'm fine with what you have too.

**Review by chcunningham — APPROVED** [review]

**Comment by chcunningham** [comment]
@padenot @aboba lgtm from me. PTAL. Note [phrasing guidance](https://github.com/w3c/webcodecs/pull/275#discussion_r657490069) (will follow in subsequent registrations).

**Comment by chcunningham** [comment]
Editors call: @aboba LGTM. @padenot, looping Mozilla H264 experts

**Comment by chcunningham** [comment]
> @padenot, looping Mozilla H264 experts

Checking in on that.

**Comment by padenot** [comment]
Let's just merge this, I went over it again and it looks fine.

**Comment by chcunningham** [comment]
Editors call: good to merge

**Pull request merged**